lib/jenkins_api_client/client.rb

```diff
@@ -145,11 +147,16 @@ def initialize(args)
       @ssl ||= false
 
       # Setting log options
-      @log_location = STDOUT unless @log_location
-      @log_level = Logger::INFO unless @log_level
-      @logger = Logger.new(@log_location)
-      @logger.level = @log_level
-
+      if @logger
+        raise ArgumentError, "logger parameter must be a Logger object" if @logger.class != Logger
```

Comment:
Let's change this check to be `.. unless @logger.is_a?(Logger)` as the logger class can be a custom logger that inherits from the `Logger` class.